### PR TITLE
CMake: further simplification

### PR DESCRIPTION
**Pull request by PikachuHy**
sub pr of https://github.com/XmacsLabs/mogan/pull/123

with the modern CMake style, we can remove redundant code.

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -5,26 +5,14 @@
...
-else (WIN32)
-  add_executable (${TeXmacs_binary_name}
-    ./Texmacs/Texmacs/texmacs.cpp
-  )
-  target_link_libraries (${TeXmacs_binary_name}
-    texmacs_body
-    ${TeXmacs_Libraries}
-    ${CMAKE_DL_LIBS}
-  )
+  set_target_properties(${TeXmacs_binary_name} PROPERTIES WIN32_EXECUTABLE TRUE)
```
[the remark this answers is not in the document]
Comment (darcy-shen) [reply src/CMakeLists.txt:14]: Here is a good github action example using Windows

https://github.com/vnotex/vnote/blob/master/.github/workflows/ci-win.yml

[the remark this answers is not in the document]
Comment (PikachuHy) [reply src/CMakeLists.txt:14]: I don't have a windows PC right now.

[the remark this answers is not in the document]
Comment (darcy-shen) [reply src/CMakeLists.txt:14]: i am using virtualbox

[the remark this answers is not in the document]
Comment (darcy-shen) [reply src/CMakeLists.txt:14]: Let me merge it now, I will make a pull request for windows user (hopefully) this weekend.

**Pull request merged**